src/In.ProjectEKA.TMHHip/DataFlow/Collect.cs

```diff
@@ -0,0 +1,536 @@
+                            careBundles.Add(careBundle);
+                        }
+
+                        break;
+                    }
+                    case HiType.DiagnosticReport:
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
+            }
+
+            var entries = new Entries(careBundles);
+            return Option.Some(entries);
+        }
+
+        private IEnumerable<CareBundle> ProcessObservationsData(DataRequest dataRequest, PatientData tmhPatientData,
+            string patientName)
+        {
+            var careBundles = new List<CareBundle>();
+            var parser = new FhirJsonParser();
+            var caseId = dataRequest.CareContexts.First().PatientReference;
+            var jsonSerializerSettings = new JsonSerializerSettings
+            {
+                Formatting = Formatting.Indented,
+                ContractResolver = new CamelCasePropertyNamesContractResolver()
+            };
+
+            if (tmhPatientData.ClinicalNotes != null)
+            {
+                var serializeObject = JsonConvert.SerializeObject(
+                    FetchClinicalNotes(dataRequest, tmhPatientData.ClinicalNotes, patientName).Result,
+                    jsonSerializerSettings);
+                var careBundle = new CareBundle(caseId, parser.Parse<Bundle>(serializeObject));
+                careBundles.Add(careBundle);
+            }
+
+            if (tmhPatientData.AllergiesData != null)
+            {
+                var serializeObject = JsonConvert.SerializeObject(
+                    FetchAllergiesData(dataRequest, tmhPatientData.AllergiesData, patientName).Result,
+                    jsonSerializerSettings);
+                var careBundle = new CareBundle(caseId, parser.Parse<Bundle>(serializeObject));
+                careBundles.Add(careBundle);
+            }
+
+            if (tmhPatientData.AbdomenExaminationsData != null)
+            {
+                var serializeObject = JsonConvert.SerializeObject(FetchAbdomenExaminationData(dataRequest,
+                    tmhPatientData.AbdomenExaminationsData, patientName).Result, jsonSerializerSettings);
+                var careBundle = new CareBundle(caseId, parser.Parse<Bundle>(serializeObject));
+                careBundles.Add(careBundle);
+            }
+
+            if (tmhPatientData.OralCavityExaminationsData != null)
+            {
+                var serializeObject = JsonConvert.SerializeObject(FetchOralCavityExaminationsData(dataRequest,
+                        tmhPatientData.OralCavityExaminationsData, patientName).Result, jsonSerializerSettings
+                );
+                var careBundle = new CareBundle(caseId, parser.Parse<Bundle>(serializeObject));
+                careBundles.Add(careBundle);
+            }
+
+            if (tmhPatientData.SurgeryHistories == null) return careBundles;
```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4940156)